### PR TITLE
SDL_rwops read/write functions return size_t again

### PR DESCRIPTION
**Pull request by slouken**
The current status is stored in the SDL_rwops 'status' field to be able to determine whether a 0 return value is caused by end of file, an error, or a non-blocking source not being ready.

The functions to read sized datatypes now return SDL_bool so you can detect read errors.

Fixes https://github.com/libsdl-org/SDL/issues/6729

### CODE REVIEW

File: docs/README-migration.md [hunk 1]
```diff
@@ -1048,6 +1017,7 @@ SDL_RWFromFP(void *fp, SDL_bool autoclose)
 }
 ```
```
Comment (madebr) [inline docs/README-migration.md:1019]: My personal preference is to return a negative value on error.
Isn't this more consistent with the other stdio functions?

Comment (slouken) [reply docs/README-migration.md:1019]: No, it's not.

**Comment by slouken** [comment]
@icculus, quick sanity check on this?

**Comment by icculus** [comment]
Things like this in SDL_RWread...


```c
    if (!context->read) {
        SDL_Unsupported();
        return 0;
    }
```

...do we want to set context->status for things like this?

**Comment by icculus** [comment]
Most of this looks good to me.

I can also see there were a million tangentially-related issues that got fixed while sorting through the code, and I love when that happens.

**Comment by slouken** [comment]
> Things like this in SDL_RWread...
> 
> ```c
>     if (!context->read) {
>         SDL_Unsupported();
>         return 0;
>     }
> ```
> 
> ...do we want to set context->status for things like this?

Oh, maybe... STATUS_UNSUPPORTED?

**Comment by slouken** [comment]
@sezero, I'm working on an sdl2-compat update for this.

**Pull request merged**